### ISSUE TITLE
Exlude all pages except those on one site, when script contains @include http* directive

### ISSUE THREAD

**Issue by superclarkk** [greasemonkey/greasemonkey#2260]
Environment: OS X 10.9.5, using Firefox 40.0
Note: apparently, it works as expected on windows, FF 41 (tested by another user)

I’m using the script,
[Mouseover Popup Image Viewer](https://greasyfork.org/en/scripts/404-mouseover-popup-image-viewer), and trying to exclude all pages, except for imdb.com.

Thus, I’ve excluded http* and included http?://*.imdb.com/* in the user configuration panel of Greasemonkey, which does not work - the script is still available to all pages (and is listed in the Greasemonkey button drop-down, tested by browsing https://www.flickr.com/explore)

Incidentally, if I change the excluded page to *, the script is available to no pages at all.

The only way to get the expected behaviour is to edit the script and change the // @include http* line to  // @exclude http* (and add the user include, as explained above, for imdb).

**Comment by janekptacijarabaci** [comment]
> Thus, I’ve excluded http* and included http?://.imdb.com/ in the user configuration panel of Greasemonkey, which does not work - the script is still available to all pages (and is listed in the Greasemonkey button drop-down, tested by browsing https://www.flickr.com/explore)
Not confirmed.

> Incidentally, if I change the excluded page to *, the script is available to no pages at all.
Yes, this is how it works.

See https://github.com/greasemonkey/greasemonkey/issues/1992
See also https://github.com/greasemonkey/greasemonkey/issues/1946

**Comment by arantius** [comment]
I have to agree, this is working as intended/designed.

**Comment by superclarkk** [comment]
@arantius I don't feel that's a satisfactory reason to close the ticket, although I agree it should be closed.

Yes, if excluded is set to *, the script isn't available to any pages. This is the expected behavior, but it wasn't the primary reason for this ticket.

To clarify, as @janekptacijarabaci has stated, the main issue has been addressed as a milestone via #1946. Having said that, I think the solution proposed in #1992 would achieve the desired result and seems easier to implement.

**Comment by tiansh** [comment]
a little complex workaround:

```
// @include     *
// @exclude     /^(?!https?:\/\/(.*\.)?(example\.com|example\.org)(\/.*)?$).*$/
```

Then the script will only works on example.com and example.org
You may change the domains to whatever you want, split multiple domains by "|".
